src/target/stm32l4.c

```diff
@@ -141,61 +243,62 @@ static void stm32l4_add_flash(target *t,
 	target_add_flash(t, f);
 }
 
-enum ID_STM32L4 {
-	ID_STM32L43  = 0x435, /* RM0394, Rev.4 */
-	ID_STM32L45  = 0x462, /* RM0394, Rev.4 */
-	ID_STM32L41  = 0x464, /* RM0394, Rev.4 */
-	ID_STM32L47  = 0x415, /* RM0351, Rev.5 */
-	ID_STM32G07  = 0x460, /* RM0444/454, Rev.1 */
-	ID_STM32L49  = 0x461, /* RM0351, Rev.5 */
-	ID_STM32L4R  = 0x470, /* RM0432, Rev.5 */
-};
-
 static bool stm32l4_attach(target *t)
 {
 	if (!cortexm_attach(t))
 		return false;
 
-	bool dual_bank = false;
-	uint32_t idcodereg = STM32L4_DBGMCU_IDCODE_PHYS;
-	uint32_t size = 0;
-
-	switch(t->idcode) {
-	case ID_STM32L47:
-	case ID_STM32L49:
-	case ID_STM32L4R:
-		dual_bank = true;
-		break;
-	case ID_STM32G07:
-		idcodereg = STM32G0_DBGMCU_IDCODE_PHYS;
-		break;
-	}
+	/* Retrive chip information, no need to check return */
+	struct stm32l4_info const *chip = stm32l4_get_chip_info(t->idcode);
+
+
+	uint32_t idcodereg = (chip->family == FAM_STM32G0x)
```

Comment:
This looks strange! You check for G0 and select something else,

Reply:
Expletives deleted.
Definitely my fault. The true and false branch should be reversed.